### DISCUSSION TITLE
Allow pause button to be held down on touchscreen to restart

### DISCUSSION THREAD

**Discussion by Awesome-Aasim** [ppy/osu#30483] [Ideas]
On keyboard, there is a quick restart option by pressing the tilde key by default. However, on a touchscreen device, one might prefer to play without the keyboard.

I suggest a button in single player mode to allow for quickly restarting by holding it down. Hold behavior for exiting play of a chart in multiplayer remains unchanged.

I would suggest:
* The "hold for menu" button is changed to "hold to restart" and has the same behavior as holding "`"/"~" on a keyboard (the key under Esc); the chart is restarted and the intro is skipped. To access the pause menu, one double taps this button instead of holding it down. "Tap again to pause"/"Hold to restart"
* Alternatively, a separate "restart" button is added to the upper left corner and has this same behavior as described.

This feature's behavior could be modified or turned on/off in settings.

**Comment by shinmorisawa** [comment]
this is a little complicated, so i don't think it will be added anytime soon.
